### PR TITLE
allow load_data_frame to handle index only dataframe

### PR DESCRIPTION
**Pull request by Hi-king**
Fixed bug that load_data_frame is currently converting index only dataframe into empty dataframe.

### CODE REVIEW

File: gokart/task.py [hunk 1]
```diff
@@ -191,21 +191,16 @@ def load_data_frame(self,
...
+        def _pd_concat(dfs):
+            if isinstance(dfs, list):
+                return pd.concat([_pd_concat(df) for df in dfs])
+            else:
+                return dfs
+        data = _pd_concat(self.load(target=target))
 
         required_columns = required_columns or set()
-        if data.empty:
+        if data.empty and len(data.index) == 0:
```
Comment (Hi-king) [inline gokart/task.py:202]: * index only dataframe should not be treated as empty one
* index only dataframe with `required_columns` should raise error

File: gokart/task.py [hunk 2]
```diff
@@ -191,21 +191,16 @@ def load_data_frame(self,
...
-                else:
-                    return dfs
-
-            data = _pd_concat(data)
+        def _pd_concat(dfs):
+            if isinstance(dfs, list):
+                return pd.concat([_pd_concat(df) for df in dfs])
+            else:
+                return dfs
+        data = _pd_concat(self.load(target=target))
```
Comment (Hi-king) [inline gokart/task.py:199]: [refactoring] just removed no use if statement

File: gokart/task.py [hunk 3]
```diff
@@ -191,21 +191,16 @@ def load_data_frame(self,
...
-                else:
-                    return dfs
-
-            data = _pd_concat(data)
+        def _flatten_recursively(dfs):
+            if isinstance(dfs, list):
+                return pd.concat([_flatten_recursively(df) for df in dfs])
+            else:
+                return dfs
+        data = _flatten_recursively(self.load(target=target))
```
Comment (Hi-king) [inline gokart/task.py:199]: [refactoring] just removed no use if statement

**Comment by vaaaaanquish** [comment]
@Hi-king LGTM

**Pull request merged**